reporter/otlp_reporter.go

```diff
@@ -731,10 +732,30 @@ func addProfileAttributes(profile *profiles.Profile,
 	indices := make([]uint64, 0, len(attributes))
 
 	addAttr := func(attr attrKeyValue) {
-		if attr.value == "" {
+		var attributeCompositeKey string
+		var attributeValue common.AnyValue
+
+		valueType := reflect.TypeOf(attr.value)
```

Comment:
Is there an option to not use `reflect`? Can we maybe use generics to avoid `reflect` here?

The reason I'm asking on a way to avoid `reflect` is, that it usually is computing intensive and code paths that use reflect will give dead code elimination strategies in the compiler a hard time.

Reply:
I couldn't make Generics work this use case, mainly because using a generic `attrKeyValue` with type constraints (string | int64) would require it to be initialized with a specific type before using it. This was creating a problem while creating sample.Attribute indices at [Line](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/2a1d9505c19e21f4a61ef4cb9571b7b21017002a/reporter/otlp_reporter.go#L654). 

So, I have used type switch to remove the `reflect`. I hope this is a viable solution.

Reply:
Thanks for the update. Switching based on the type information is a good replacement in this case to not use `reflect` here :+1: